openpype/hosts/maya/plugins/publish/collect_arnold_scene_source.py

```diff
@@ -21,18 +22,20 @@ def process(self, instance):
                 self.log.warning("Skipped empty instance: \"%s\" " % objset)
                 continue
             if objset.endswith("content_SET"):
-                instance.data["setMembers"] = cmds.ls(members, long=True)
-                self.log.debug("content members: {}".format(members))
+                set_members = get_all_children(cmds.ls(members, long=True))
```

Comment:
I believe this could just be `get_all_children(members)` - if not, then there's already something off with the members input since those should already have unique paths (not necessarily long paths).

It's good to note that `get_all_children` includes all children, but not the original members themselves. So it only includes the children of `members` not the original `members`.

You might want to do:
```python
members = cmds.ls(members, long=True)
children = get_all_children(members)
hierarchy = list(set(members + children))
```

_It is technically possible for some of the members to be present in the result of `get_all_children` if one of the members is a child of another member._